lib/rubygems/commands/setup_command.rb

```diff
@@ -17,7 +17,7 @@ def initialize
 
     super 'setup', 'Install RubyGems',
           :format_executable => true, :document => %w[ri],
-          :force => false,
+          :force => true,
```

Comment:
I'm still against to set the default as `--force` option. The destructive action should be set by end-user explicity.

Can you add the warning message with target executables? like this:

```
WARNING: rubygems-update will overwrite the following files:
/path/to/bundle
```

Reply:
@hsbt I understand your concern but the point of this PR is to not change the previous behavior. We can think of changing the default in the future (4.0?) but for now the main priority IMO is to restore the previous behavior.

See how things work in rubygems 3.0:

```
$ gem --version
3.0.5
$ echo "puts 'EDIT'" >> /home/deivid/.rbenv/versions/2.6.5/bin/bundle
$ cat /home/deivid/.rbenv/versions/2.6.5/bin/bundle
#!/home/deivid/.rbenv/versions/2.6.5/bin/ruby
#
# This file was generated by RubyGems.
#
# The application 'bundler' is installed as part of a gem, and
# this file is here to facilitate running it.
#

require 'rubygems'

version = ">= 0.a"

str = ARGV.first
if str
  str = str.b[/\A_(.*)_\z/, 1]
  if str and Gem::Version.correct?(str)
    version = str
    ARGV.shift
  end
end

if Gem.respond_to?(:activate_bin_path)
load Gem.activate_bin_path('bundler', 'bundle', version)
else
gem "bundler", version
load Gem.bin_path("bundler", "bundle", version)
end
puts 'EDIT'
deivid@chaba ~ $ gem update --system 3.0.6
Updating rubygems-update
Successfully installed rubygems-update-3.0.6
Installing RubyGems 3.0.6
Bundler 1.17.3 installed
RubyGems 3.0.6 installed
Regenerating binstubs

=== 3.0.6 / 2019-08-17

Bug fixes:

* Revert #2813. It broke the compatibility with 3.0.x versions.

=== 3.0.5 / 2019-08-16

Minor enhancements:

* Use env var to configure api key on push. Pull request #2559 by Luis
  Sagastume.
* Unswallow uninstall error. Pull request #2707 by David Rodríguez.
* Expose windows path normalization utility. Pull request #2767 by David
  Rodríguez.
* Clean which command. Pull request #2801 by Luis Sagastume.
* Upgrading S3 source signature to AWS SigV4. Pull request #2807 by
  Alexander Pakulov.
* Remove missleading comment, no reason to move Gem.host to Gem::Util.
  Pull request #2811 by Luis Sagastume.
* Drop support for 'gem env packageversion'. Pull request #2813 by Luis
  Sagastume.
* Take into account just git tracked files in update_manifest rake task.
  Pull request #2816 by Luis Sagastume.
* Remove TODO comment, there's no Gem::Dirs constant. Pull request #2819
  by Luis Sagastume.
* Remove unused 'raise' from test_case. Pull request #2820 by Luis
  Sagastume.
* Move TODO comment to an information comment. Pull request #2821 by Luis
  Sagastume.
* Use File#open instead of Kernel#open in stub_specification.rb. Pull
  request #2834 by Luis Sagastume.
* Make error code a gemcutter_utilities a constant. Pull request #2844 by
  Luis Sagastume.
* Remove FIXME comment related to PathSupport. Pull request #2854 by Luis
  Sagastume.
* Use gsub with Hash. Pull request #2860 by Kazuhiro NISHIYAMA.
* Use the standard RUBY_ENGINE_VERSION instead of JRUBY_VERSION. Pull
  request #2864 by Benoit Daloze.
* Do not mutate uri.query during s3 signature creation. Pull request #2874
  by Alexander Pakulov.
* Fixup #2844. Pull request #2878 by SHIBATA Hiroshi.

Bug fixes:

* Fix intermittent test error on Appveyor & Travis. Pull request #2568 by
  MSP-Greg.
* Extend timeout on assert_self_install_permissions. Pull request #2605 by
  SHIBATA Hiroshi.
* Better folder assertions. Pull request #2644 by David Rodríguez.
* Fix default gem executable installation when folder is not `bin/`. Pull
  request #2649 by David Rodríguez.
* Fix gem uninstall behavior. Pull request #2663 by Luis Sagastume.
* Fix for large values in UID/GID fields in tar archives. Pull request
  #2780 by Alexey Shein.
* Fixed task order for release. Pull request #2792 by SHIBATA Hiroshi.
* Ignore GEMRC variable for test suite. Pull request #2837 by SHIBATA
  Hiroshi.


------------------------------------------------------------------------------

RubyGems installed the following executables:
	/home/deivid/.rbenv/versions/2.6.5/bin/gem
	/home/deivid/.rbenv/versions/2.6.5/bin/bundle

RubyGems system software updated
$ cat /home/deivid/.rbenv/versions/2.6.5/bin/bundle
#!/home/deivid/.rbenv/versions/2.6.5/bin/ruby
#
# This file was generated by RubyGems.
#
# The application 'bundler' is installed as part of a gem, and
# this file is here to facilitate running it.
#

require 'rubygems'

version = ">= 0.a"

str = ARGV.first
if str
  str = str.b[/\A_(.*)_\z/, 1]
  if str and Gem::Version.correct?(str)
    version = str
    ARGV.shift
  end
end

if Gem.respond_to?(:activate_bin_path)
load Gem.activate_bin_path('bundler', 'bundle', version)
else
gem "bundler", version
load Gem.bin_path("bundler", "bundle", version)
end
```

So, rubygems 3.0 forcefully overwrites executables and already informs clearly about the executables it installed.

However, 3.1, even if with this PR also overwrites executables, it no longer informs about the `bundler` executables being written. I think that's an issue and I will fix it as you suggested :+1:.

Reply:
I understand to important to keep the previous behavior. But also the previous behavior is not expected specification.

Ok, I will merge this for your suggestion.

Reply:
I was fixing the bug you found: Installed `bundle` executable is no longer informed in rubygems 3.1. I will open a separate PR since we are already merging this.

Reply:
See https://github.com/rubygems/rubygems/pull/3042.

Reply:
👍